formats/label-to-vector-plugin/src/polus/plugins/formats/label_to_vector/__main__.py

```diff
@@ -0,0 +1,228 @@
+)
+logger = helpers.make_logger("polus.plugins.formats.label_to_vector")
+
+app = typer.Typer()
+
+
+def flow_thread(
+    file_name: pathlib.Path,
+    zarr_path: pathlib.Path,
+    coordinates: tuple[int, int, typing.Optional[int]],
+) -> bool:
+    """Calculates the flows on a tile and saves it to the zarr file.
+
+    This is designed to run in a thread.
+
+    Args:
+        file_name: The file to read the tile from.
+        zarr_path: The zarr file to save the tile to.
+        coordinates: The coordinates of the tile to read.
+
+    Returns:
+        True if the tile was successfully processed.
+    """
+    x, y, z = coordinates
+    ndims = 2 if z is None else 3
+    z = 0 if z is None else z
+
+    # Load the data
+    with bfio.BioReader(file_name) as reader:
+        x_shape, y_shape, z_shape = reader.X, reader.Y, reader.Z
+
+        x_min = max(0, x - constants.TILE_OVERLAP)
+        x_max = min(x_shape, x + constants.TILE_SIZE + constants.TILE_OVERLAP)
+
+        y_min = max(0, y - constants.TILE_OVERLAP)
+        y_max = min(y_shape, y + constants.TILE_SIZE + constants.TILE_OVERLAP)
+
+        z_min = max(0, z - constants.TILE_OVERLAP)
+        z_max = min(z_shape, z + constants.TILE_SIZE + constants.TILE_OVERLAP)
+
+        masks = numpy.squeeze(reader[y_min:y_max, x_min:x_max, z_min:z_max, 0, 0])
+
+    masks = masks if ndims == 2 else numpy.transpose(masks, (2, 0, 1))  # noqa: PLR2004
+    masks_shape = masks.shape
+
+    # Calculate index and offsets
+    x_overlap = x - x_min
+    x_min, x_max = x, min(x_shape, x + constants.TILE_SIZE)
+    cx_min, cx_max = x_overlap, x_max - x_min + x_overlap
+
+    y_overlap = y - y_min
+    y_min, y_max = y, min(y_shape, y + constants.TILE_SIZE)
+    cy_min, cy_max = y_overlap, y_max - y_min + y_overlap
+
+    z_overlap = z - z_min
+    z_min, z_max = z, min(z_shape, z + constants.TILE_SIZE)
+    cz_min, cz_max = z_overlap, z_max - z_min + z_overlap
+
+    # Save the mask before transforming
+    if ndims == 2:  # noqa: PLR2004
+        masks_original = masks[numpy.newaxis, numpy.newaxis, numpy.newaxis, :, :]
+    else:
+        masks_original = masks[numpy.newaxis, numpy.newaxis, :, :]
+    masks_original = masks_original[:, :, cz_min:cz_max, cy_min:cy_max, cx_min:cx_max]
+
+    # noinspection PyTypeChecker
+    zarr_root = zarr.open(str(zarr_path))[0]
+    zarr_root[0:1, 0:1, z_min:z_max, y_min:y_max, x_min:x_max] = numpy.asarray(
+        masks_original != 0,
+        dtype=numpy.float32,
+    )
+    zarr_root[
+        0:1,
+        ndims + 1 : ndims + 2,
+        z_min:z_max,
+        y_min:y_max,
+        x_min:x_max,
+    ] = numpy.asarray(masks_original, dtype=numpy.float32)
+
+    if not numpy.any(masks):
+        logger.debug(
+            f"Tile {(x, y, z) = } in {file_name.name} has no objects. "
+            f"Setting flows to zero...",
+        )
+        flows = numpy.zeros((ndims, *masks.shape), dtype=numpy.float32)
+    else:
+        # Normalize
```

Comment:
this is not obvious without additional hints why we need to do that rather than just keep the original labels...  The comment says "Normalize"? It seems that this code can be removed altogether since it is also done in `convert`